libsplinter/src/matrix.rs

```diff
@@ -46,20 +64,67 @@ impl Envelope {
 /// MatrixLifeCycle trait abstracts out adding and removing connections to a
 /// connection handler without requiring knowledge about sending or receiving messges.
 pub trait MatrixLifeCycle: Clone + Send {
+    /// Add a connection to the connection handler with a connection ID
+    ///
+    /// # Arguments
+    ///
+    /// * `connection` - A boxed connection that will be passed to the connection handler.
+    /// * `id` - a unique connection ID that will be used to sent messages over a connection.
+    ///
+    /// If the add failed, a MatrixAddError will be returned.
     fn add(&self, connection: Box<dyn Connection>, id: String) -> Result<usize, MatrixAddError>;
+
+    // Removes a connection from the connection handler with a connection ID.
+    //
+    /// # Arguments
+    ///
+    /// * `id` - the unique connection ID for the connection that should be removed
+    ///
+    /// If the remove failed, a MatrixRemoveError will be returned.
     fn remove(&self, id: &str) -> Result<Box<dyn Connection>, MatrixRemoveError>;
 }
 
+/// MatrixSender trait abstracts out sending messages through a connection handler without
+/// requiring knowledge adding and removing connections, and receiving messages
 pub trait MatrixSender: Clone + Send {
+    /// Send a message over the specified connection.
+    ///
+    /// # Arguments
+    ///
+    /// * `id` - the unique connection ID the message should be sent over
+    /// * `message` - the bytes of the messages
+    ///
+    /// If the send failed, a MatrixSendError will be returned.
     fn send(&self, id: String, message: Vec<u8>) -> Result<(), MatrixSendError>;
 }
 
+/// MatrixReceiver trait abstracts out receiving messages from a connection handler without
+/// requiring knowledge adding and removing connections, and sending messages
 pub trait MatrixReceiver: Clone + Send {
+    /// Attempt to receive a message from the connection handler. This function will block until
```

Comment:
Also would be sure to note here that the message will come from the first ready connection detected.

Reply:
added